sherpa/astro/datastack/ds.py

```diff
@@ -108,47 +121,70 @@ def clear_stack(self):
 
-        A brief summary of each data set in the stack is printed to the
-        standard output channel. The information displayed depends on the
-        type of each data set.
-        """
+        For each dataset in the datastack, id, name, and a number of
+        keys from the header (e.g. TELESCOP, MJD-OBS) are
+        collected. Then columns that are identical for all datasets
+        are removed (e.g. if no dataset has a GRATING keyword). That
+        way, the output can depend on the datasets loaded. In most
+        cases, only a few of a relatively long list of keywords will
+        differ between datasets, printing out only the relevant
+        information.
+
+        Returns
+        -------
+        header : list of strings
+            Column names
+        data : list of lists
+            Each list is a row, corresponding to one dataset
 
-        # The observation-date (MJD) format keyword was MJD_OBS but
-        # has changed (CIAO 4.12) for Chandra data to MJD-OBS, so
-        # need to look for both. The idea is to use the new key
-        # and then fall back to the old one.
-        #
+        '''
+        tab = []
         for dataset in self.filter_datasets():
-            obsid = "N/A"
-            time = "N/A"
-            timekey = 'MJD-OBS'
+            row = [dataset['id'], dataset['data'].name]
             try:
                 hdr = dataset['data'].header
+                row.extend([hdr.get(k, '') for k in keys_to_display])
+            except AttributeError:
+                row.append([''] * len(keys_to_display))
+            tab.append(row)
 
-                try:
-                    obsid = hdr['OBS_ID']
-                except KeyError:
-                    pass
-
-                for key in ['MJD-OBS', 'MJD_OBS']:
-                    try:
-                        time = hdr[key]
-                        timekey = key
-                        break
-                    except KeyError:
-                        pass
+        if len(tab) == 0:
+            header = ['empty datastack']
+            data = numpy.array([])
+        else:
+            data = numpy.array(tab)
+            # Find columns where not all entries are identical
+            ind = ~numpy.all(data == data[0, :], axis=0)
+            # But always keep the first two (id, name)
+            ind[:2] = True
+
+            header = numpy.array(['id', 'name'] + keys_to_display)[ind]
+            data = data[:, ind]
+        return header, data
+
+    def _repr_html_(self):
+        """Return a HTML (string) representation of the stack
+        """
+        header, data = self._summary_for_display()
+        htab = html_table(header, data, rowcount=False,
+                          summary=f'datastack with {data.shape[0]} datasets')
+        return html_from_sections(self, htab)
 
-            except AttributeError:
-                pass
+    # QUS: should this use the logger instead of print?
```

Comment:
The existing `show_xxx` routines all use `sherpa.utils.send_to_pager` [at least those in sherpa[.astro].ui.utils do] so we should probly do the same.  I'm not a great fan of `send_to_pager` but its what we do...

Reply:
Of course, I don't know how the `show_xxx` routines work in the noteook/lab interface.

Reply:
This is not new code, it's merely moved to a different line: https://github.com/sherpa/sherpa/pull/1194/files#diff-1b2333a505749af538cd4ba44a468fce606ee4cea032f1e6544ce66aa85b3debL111

This `shows_stack` function did not `send_to_pager` before and it does not do so now. All I have changed in the content of the message. (And there is some refactoring, because `_repr_html` needs similar information.
So, just to clarify: Do you think I should bring is in-line with what the other `show_xxx` functions do because I'm touching this function now?
I don't know what the original motivations was to use or not use `send_to_pager` in `show_stack`.